### PR TITLE
check for subscriptionDetails

### PR DESCRIPTION
**Pull request by keinagae**
added check to prevent crash if subscriptionDetails is null

### CODE REVIEW

File: src/android/IabHelper.java [hunk 1]
```diff
@@ -491,10 +491,14 @@ public void onProductDetailsResponse(BillingResult billingResult, List<ProductDe
...
-                           .get(selectedOfferIndex)
-                           .getOfferToken();
+                        String offerToken="";
+                        List<ProductDetails.SubscriptionOfferDetails> subscriptionDetails = productDetails
+                            .getSubscriptionOfferDetails();
+                        if (subscriptionDetails!=null){
+                            offerToken = subscriptionDetails
+                              .get(selectedOfferIndex)
+                              .getOfferToken();
+                         }
```
Comment (rmoehn) [inline src/android/IabHelper.java:501]: When `subscriptionDetails` is `null`, `.setOfferToken` will be passed an empty string below. What is the effect of that?

Comment (keinagae) [reply src/android/IabHelper.java:501]: it wont effect at all as previously if getSubscriptionOfferDetails was null then it was generating  an exception on .get(selectedOfferIndex) and was not returning any response but i tested if offerToken is blank then it does not generate any error and work as expected so i left it like that but let me if you want me to change it

Comment (rmoehn) [reply src/android/IabHelper.java:501]: I'm not the maintainer, so my opinion isn't that important. But setting the offer token to an empty string feels wrong. Who knows what other code is dealing with that offer token later (perhaps in cases that you haven't tested) and gets confused by the empty string?

Comment (keinagae) [reply src/android/IabHelper.java:501]: offer token is not being used any where else in this method

Comment (rmoehn) [reply src/android/IabHelper.java:501]: Not in this method, no. But it is passed to Google SDK code with `.setOfferToken` and then who knows how the Google code uses it.

Comment (keinagae) [reply src/android/IabHelper.java:501]: on latest commit changed it to better form let me know if its oky

**Comment by manuelwenner** [comment]
I don't know which of the two commits but with this version of `IabHelper.java` In-App-Purchases are working in my old cordova project 🙏🥳  Would be great if it would be merged soon.

File: src/android/IabHelper.java (hunk 1 above)
Comment (rmoehn) [reply src/android/IabHelper.java:501]: It's better. Thanks! Of course, I'm not the maintainer. @wccrawford, the owner of this repo, appears to make changes occasionally. Let's hope he takes into account your suggestion.

**Review by wccrawford — APPROVED** [review]

**Comment by wccrawford** [comment]
Thanks for working on this!  I'm still a little unsure of how this works if Google's API doesn't return the information needed, but I'll take your word that it's working in your situation.  I haven't encountered it myself.

**Comment by wccrawford** [comment]
Also, thanks @rmoehn for reviewing this change and helping improve it!

**Comment by manuelwenner** [comment]
@wccrawford @rmoehn I use Firebase in my project and since the last update based on this plugin i get the following error/problems.
Screenshot 2023-01-04 at 14 41 03

**Comment by rmoehn** [comment]
From which version/commit to which version/commit was this ‘last update’?

**Pull request merged**